src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx

```diff
@@ -780,7 +797,7 @@ calc_emission_kernel(const double current_alpha_estimate_zyx,
                                     sigma_dp*sigma_dp,
                                     distance_dzdydx*distance_dzdydx,
                                     current_alpha_estimate_zyx*current_alpha_estimate_zyx)
-    : calc_kernel_from_precalculated((*kmnorm_sptr)[0][l][m],
+    : calc_kernel_from_precalculated((*kpnorm_sptr)[0][l][m],
```

Comment:
I didn't check, but is the `[0]` index apropriate?

Reply:
I believe this is a weird hackaround - an image type (i.e., 3D) is used for a matrix (2D). I think this should probably be refactored and just a 2D datatype used. But I think there are a few things that could be refactored in this file, so I'd flag this for there.

Reply:
Yes this was done because there were problems with DiscretisedDensity if I used a 2D array. So I had to use 3D but only fill the elements [0][l][m]